File: webapp/airTicket/src/app/core/service/flight-management.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {FlightManagement} from '../../entity/FlightManagement';
import {Observable} from 'rxjs';
import {Pageable} from '../../entity/norm/pageable';
import {Page} from '../../entity/norm/page';

@Injectable({
  providedIn: 'root'
})
export class FlightManagementService {

  baseUrl = 'api/flightManagement';

  constructor(private httpClient: HttpClient) {
  }

  // 保存
  save(flightManagement: FlightManagement): Observable<FlightManagement> {
    return this.httpClient.post<FlightManagement>(this.baseUrl, flightManagement);
  }

  //
  pageByStartingPlaceAndDestinationStartTime(startingPlaceId: number, destinationId: number, startTime: Date, pageable: Pageable): Observable<Page<FlightManagement>> {
    const params: any = {
      size: pageable.size.toString(),
      page: pageable.page.toString(),
    };
    if (startingPlaceId) {
      params.startingPlaceId = startingPlaceId.toString();
    }
    if (destinationId) {
      params.destinationId = destinationId.toString();
    }
    if (startTime) {
      params.startTime = startTime.getTime().toString();
    }
    return this.httpClient.get<Page<FlightManagement>>(this.baseUrl + '/page', {params});
  }

  // 通过id获取
  getById(id: number): Observable<FlightManagement> {
    return this.httpClient.get<FlightManagement>(this.baseUrl + '/' + id);
  }

  delete(id: number): Observable<any> {
    return this.httpClient.delete(this.baseUrl + '/' + id);
  }

  update(id: number, management: FlightManagement): Observable<FlightManagement> {
    return this.httpClient.put<FlightManagement>(this.baseUrl + '/' + id, management);
  }
}

File: api/airTicket/src/main/java/com/xiang/airTicket/service/FlightManagementService.java
package com.xiang.airTicket.service;

import com.xiang.airTicket.entity.FlightManagement;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Date;
import java.util.List;

public interface FlightManagementService {

    // 保存
    FlightManagement save(FlightManagement flightManagement);

    //  通过始发地 目的地 起飞日期获取分页数据
    Page<FlightManagement> pageByStartingPlaceAndDestinationStartTime(
            Long startingPlaceId,
            Long destinationId,
            Date startTime,
            Pageable pageable
    );

    //  通过始发地 目的地 起飞日期获取分页数据 没有结束日期限制
    Page<FlightManagement> pageByStartingPlaceAndDestinationStartTimeGetContent(
            Long startingPlaceId,
            Long destinationId,
            Date startTime,
            Pageable pageable
    );

    List searchFlight(Long startingPlaceId,
                      Long destinationId,
                      Date startTime);

    // 更新
    FlightManagement update(Long id, FlightManagement flightManagement);

    // 通过id获取
    FlightManagement getById(Long id);

    // 删除
    void delete(Long id);

}
